Implementing connection recovery via a web client (#4, #5)

tyranron opened the pull request:
Continued from #4, #5

> Implemented a call to the remote method connect, after an abnormal web socket connection close.
> 
> Additionally:
> 
> It is obvious to us that not all cases of closing a web-socket connection require the recovery of a connection. So in most cases, if the web-socket connection is closed by the server, attempts to restore the web-socket connection are not needed. This happens when a broadcast participant is disabled on purpose.
> 
> To handle these cases implemented processing of closing status of a web connection. According to the spec, statuses with a code of less than 4000 are reserved for service needs, and more than 4000 can be processed as user-defined. Currently several statuses are used:
> 
> 4001 - Reconnection fault;
> 4002 - RPC session has been closed by js client application.
> 4003 - RPC request is canceled;
> The differentiation of the behavior of the WebSocketWithReconnect object is implemented depending on the closing status of the web-socket connection. If the close status code is more than 4000, WebSocketWithReconnect attempts to recovery the closed web-socket connection, and does not attempt to recovery the web-socket connection otherwise.

tyranron committed:
Diff from 'flexconstructor:reconnect' branch

flexconstructor committed:
Small corrections

flexconstructor commented:
@tyranron , FCM:

```
Implement connection recovery via a web client (#6)

- update webSocketWithReconnect js
- update jsonrpcclient js
- update Openvidu ts
- update Session ts
- add RPC ID to Connection ts
```

tyranron commented:
@flexconstructor do not forget to use code block for FCM.

flexconstructor commented:
@tyranron , Ok, sorry... Is content of the message right?

flexconstructor committed:
Implement connection recovery via a web client (#6)

- update webSocketWithReconnect js
- update jsonrpcclient js
- update Openvidu ts
- update Session ts
- add RPC ID to Connection ts
